src/apps/content-editor/src/app/components/Editor/Field/Field.js

```diff
@@ -402,20 +410,49 @@ export default function Field({
 
     case "yes_no":
       if (settings.options) {
-        const binaryFieldOpts = Object.values(settings.options);
-
         return (
-          <FieldTypeBinary
-            name={name}
-            label={FieldTypeLabel}
-            description={description}
-            tooltip={settings.tooltip}
-            required={required}
-            value={value}
-            onChange={onChange}
-            offValue={binaryFieldOpts[0]}
-            onValue={binaryFieldOpts[1]}
-          />
+          <>
```

Comment:
This feels like a lot to have to write any time we need a `FieldTypeBinary`. @agalin920 should this get rolled up and provide a simpler interface?

Reply:
Granted @d88naimi has 2 other PRs with this structure implemented. @d88naimi what's your time estimate on how long these 3 PRs took?

Reply:
Something to consider. Even if it is quick to implement with some much structure and plumbing how much are the odds of something being done wrong increased?

Reply:
@shrunyan all PRs took maybe 3 days with testing (10 FieldTypeBinary converted).

I estimate though now removing all the null checks and lifting that logic to a different location with the help of @agalin920 another 2-3 days though